Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/Clusterer.h

```diff
@@ -90,8 +92,59 @@ class Clusterer
   }
 
  private:
-  enum { kMaxRow = 650 }; // Anything larger than the real number of rows (512 for ALPIDE)
   void initChip(UInt_t first);
+
+  ///< add new precluster at given row of current column for the fired pixel with index ip in the ChipPixelData
+  void addNewPrecluster(UInt_t ip, UShort_t row)
+  {
+    mPreClusterHeads.push_back(mPixels.size());
+    // new head does not point yet (-1) on other pixels, store just the entry of the pixel in the ChipPixelData
+    mPixels.emplace_back(-1, ip);
+    int lastIndex = mPreClusterIndices.size();
+    mPreClusterIndices.push_back(lastIndex);
+    mCurr[row] = lastIndex; // store index of the new precluster in the current column buffer
+  }
+
+  ///< add cluster at row (entry ip in the ChipPixeData) to the precluster with given index
+  void expandPreCluster(UInt_t ip, UShort_t row, int preClusIndex)
+  {
+    auto& firstIndex = mPreClusterHeads[mPreClusterIndices[preClusIndex]];
+    mPixels.emplace_back(firstIndex, ip);
+    firstIndex = mPixels.size() - 1;
+    mCurr[row] = preClusIndex;
+  }
+
+  ///< recalculate min max row and column of the cluster accounting for the position of pix
+  void adjustBoundingBox(const o2::ITSMFT::PixelData pix, UShort_t& rMin, UShort_t& rMax,
+                         UShort_t& cMin, UShort_t& cMax) const
+  {
+    if (pix.getRowDirect() < rMin) {
+      rMin = pix.getRowDirect();
+    } else if (pix.getRowDirect() > rMax) {
+      rMax = pix.getRowDirect();
+    }
+    if (pix.getCol() < cMin) {
+      cMin = pix.getCol();
+    } else if (pix.getCol() > cMax) {
+      cMax = pix.getCol();
+    }
```

Comment:
The comparisons with rMin, rMax, cMin and cMax must be independent.  No "else" between the if's.